apollo-federation/src/sources/connect/spec/directives.rs

```diff
@@ -102,38 +113,41 @@ impl TryFrom<&Component<Directive>> for SourceDirectiveArguments {
             let arg_name = arg.name.as_str();
 
             if arg_name == SOURCE_NAME_ARGUMENT_NAME.as_str() {
-                name = Some(
-                    arg.value
-                        .as_str()
-                        .expect("`name` field in `@source` directive is not a string"),
-                );
+                name = Some(arg.value.as_str().ok_or_else(|| {
```

Comment:
When your closure just returns an error like this, you can use `ok_or` instead of `ok_or_else` to just directly map to that error. Applies to many of the other `ok_or_else` calls as well.

Reply:
switched to ok_or with a macro